src/managers/k8s.py

```diff
@@ -0,0 +1,215 @@
+            ),
+            spec=ServiceSpec(
+                externalTrafficPolicy="Local",
+                type="LoadBalancer",
+                selector={"app.kubernetes.io/name": self.app_name},
+                ports=ports,
+            ),
+        )
+
+    def get_pod(self, pod_name: str) -> Pod:
+        """Gets the Pod via the K8s API."""
+        return self.client.get(
+            res=Pod,
+            name=pod_name,
+        )
+
+    def get_service(self, service_name: str) -> Service:
+        """Gets the Service via the K8s API."""
+        return self.client.get(
+            res=Service,
+            name=service_name,
+        )
+
+    def get_node(self, pod_name: str) -> Node:
+        """Gets the Node the Pod is running on via the K8s API."""
+        pod = self.get_pod(pod_name)
+        if not pod.spec or not pod.spec.nodeName:
+            raise Exception("Could not find podSpec or nodeName")
+
+        return self.client.get(
+            Node,
+            name=pod.spec.nodeName,
+        )
+
+    def get_node_ip(self, pod_name: str) -> str:
+        """Gets the IP Address of the Node of a given Pod via the K8s API."""
+        try:
+            node = self.get_node(pod_name)
+        except ApiError as e:
+            if e.status.code == 403:
+                return ""
+
+        if not node.status or not node.status.addresses:
+            raise Exception(f"No status found for {node}")
+
+        for addresses in node.status.addresses:
+            if addresses.type in ["ExternalIP", "InternalIP", "Hostname"]:
+                return addresses.address
+
+        return ""
+
+    def get_nodeport(self, auth: Literal["plain", "tls"]) -> int:
+        """Gets the NodePort number for the service via the K8s API."""
+        if not (service := self.get_service(self.exposer_service_name)):
+            raise Exception("Unable to find Service")
+
+        if not service.spec or not service.spec.ports:
+            raise Exception("Could not find Service spec or ports")
+
+        for port in service.spec.ports:
+            if str(port.name).endswith(auth):
+                return port.nodePort
+
+        raise Exception(f"Unable to find NodePort using {auth} for the {service} service")
+
+    def get_loadbalancer(self) -> str:
+        """Gets the LoadBalancer address for the service via the K8s API."""
+        if not (service := self.get_service(self.exposer_service_name)):
+            raise Exception("Unable to find Service")
+
+        if (
+            not service.status
+            or not (lb_status := service.status.loadBalancer)
+            or not lb_status.ingress
+        ):
+            raise Exception("Could not find Service status or LoadBalancer")
+
+        lb: LoadBalancerIngress
+        for lb in lb_status.ingress:
+            if lb.ip is not None:
+                return lb.ip
+
+        raise Exception(f"Unable to find LoadBalancer ingress for the {service} service")
```

Comment:
**question:** Although impractical to int-test, have we validated this PR against a public-cloud LB? Say on EKS or GKE?

Reply:
Not yet, that would be a different ticket, namely DPE-5988.
In addition, there is a new `loadbalancer-extra-annotations` configuration option mentioned in the specs doc DA122 if we need to make some LB adjustments for AKS